HEATDamage.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Text;
+using System.Diagnostics;
+
+namespace PhysicalDamage.Core
+{
+    class HEATDamage:Damage
+    {
+        public struct HEATData
+        {
+            public float Pierce;
+
+            public float ArmorHEATFactor;
+
+            public float HealthHEATFactor;
+        }
+
+        private HEATData data;
+
+        public HEATDamage(HEATData d,Damage.Target t):base(EDamageTypes.HEAT, t)
+        {
+            data = d;
+        }
+
+        override public float DealDamage()
```

Comment:
Consider renaming this to `CalculateDamage()` or something along the lines of gathering rather than *giving*. `DealDamage` does update other variables but doesn't really make that clear. A class or struct could be created to return the updated ERA, updated Armor, and final damage. 

Here's my shitty example to illustrate:
`
public struct Damage
{
   /* ... Code here ... */
  public Damage(params here) {
     /* Similar to a class. */
  }
}
`